aten/src/ATen/native/mps/OperationUtils.h

```diff
@@ -382,23 +382,92 @@ class MetalShaderLibrary {
   std::unordered_map<std::string, std::pair<id<MTLComputePipelineState>, id<MTLFunction>>> cplMap;
 };
 
+namespace detail {
+template <typename T>
+class has_size_type {
```

Comment:
nit; Could probably a bit cleaner with constexpr if now.

Reply:
Do you have an example in mind? (I've tried doing it without SFINAE, but couldn't figure out how to)